subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -1722,16 +1723,17 @@ int ull_ticker_stop_with_mark(uint8_t ticker_handle, void *param,
 		return -EALREADY;
 	}
 
-	ret = ull_disable(lll_disable);
-	if (ret) {
-		return -EBUSY;
-	}
+	err = ull_disable(lll_disable);
 
 	mark = ull_disable_unmark(param);
 	if (mark != param) {
 		return -ENOLCK;
 	}
 
+	if (err && (err != -EALREADY)) {
+		return err;
+	}
+
```

Comment:
In this change you've change the call to `ull_disable_unmark` to be done onconditionally as the only place in the code base. Is that intentional? 

Later I see that you do a similar change where it is still called if there's an error, but only if the error is `EALREADY`. Should that have been done here as well?

Reply:
Resolved outside of Github :)

Reply:
`ull_disable_mark` and `ull_disable_unmark` are used to save the current state/role context pointer when ticker instance is being stopped so that the ticker operation callback can detect a race when both thread and ULL execution context try to call `ticker_stop()`.

Here at line 1733, if `ull_disable()` returns `-EALREADY` then the error is returned back to the caller of `ull_ticker_stop_with_mark()`.

Whereas in line 2590, if `ull_disable()` returns `-EALREADY` then we ignore it as `disable()` which is the caller is directly called by HCI interface to disable the advertiser.

In other words,

In the first instance, `ull_ticker_stop_with_mark()` function is reused across the Controller and the error code is returned back to caller.

In the second instance, `disable()` function is direct call from HCI interface to disable advertising and error (except `-EALREADY` ) return from ull_disable() is design fault, hence use of LL_ASSERT